storage/log_storage.go

```diff
@@ -124,6 +125,30 @@ type LogStorage interface {
 	// Duplicates are only reported if the underlying tree does not permit duplicates, and are
 	// considered duplicate if their leaf.LeafIdentityHash matches.
 	QueueLeaves(ctx context.Context, treeID int64, leaves []*trillian.LogLeaf, queueTimestamp time.Time) ([]*trillian.QueuedLogLeaf, error)
+
+	// AddSequencedLeaves stores the `leaves` and associates them with the log
+	// positions according to their `LeafIndex` field. The indices must be unique
+	// and sorted in ascending order.
+	//
+	// If error is nil, the returned slice is the same size as the input, entries
+	// correspond to the `leaves` in the same order. Each entry describes the
+	// result of adding the corresponding leaf.
+	//
+	// Possible `QueuedLogLeaf.status` values with their semantics:
+	//  - OK: The leaf has been successfully stored.
+	//  - AlreadyExists: The storage has another leaf with the same `LeafIndex`.
```

Comment:
"the storage already contains an identical leaf at the specified `LeafIndex`."?

Reply:
Done.